lib/src/local_working_copy.rs

```diff
@@ -863,7 +863,7 @@ impl TreeState {
                         .collect_vec()
                 });
 
-                Some(Box::new(PrefixMatcher::new(&repo_paths)))
+                Some(Box::new(FilesMatcher::new(&repo_paths)))
```

Comment:
- I don't know if Watchman returns directories, actually; https://facebook.github.io/watchman/docs/cmd/since only mentions files. Presumably it would trigger in a case where the metadata for the directory changed, which I guess we wouldn't care about.
- IIRC I wrote it like this because (I had thought that) `FilesMatcher` won't work if we're given `foo/bar/baz`, because we'll never reach `foo/bar` to begin with. So I had written `PrefixMatcher` here to ensure that we visit `foo` and then `foo/bar` so that we can actually reach the target file `foo/bar/baz`. I could be mistaken about this; if so, then we can keep `FilesMatcher`.

Reply:
> Presumably it would trigger in a case where the metadata for the directory changed

Yes. `jj debug watchman query-changed-files --ignore-working-copy` includes `lib/src` and `lib/src/x` if the file `x` is added. The directory is also included if `touch`ed.

> `FilesMatcher` won't work if we're given `foo/bar/baz`

In that case, `FilesMatcher` visits all parent directories, so it should work.